app/src/main/kotlin/pl/gov/mc/protego/ProtegoApp.kt

```diff
@@ -1,19 +1,35 @@
 package pl.gov.mc.protego
 
 import android.app.Application
+import android.content.Context
 import org.koin.android.ext.koin.androidContext
 import org.koin.core.context.startKoin
+import pl.gov.mc.protego.bluetooth.BeaconIdManager
+import pl.gov.mc.protego.bluetooth.BluetoothBeaconIdExchangeManager
 import pl.gov.mc.protego.di.*
 import timber.log.Timber
 
+/*
+TODO:
+1. Czy przerywac skanowanie na czas połączenia? Rozgłaszanie?
+ */
+
 class ProtegoApp : Application() {
 
+    companion object {
+        lateinit var context: Context
+    }
+    private lateinit var m: BluetoothBeaconIdExchangeManager
+
     override fun onCreate() {
         super.onCreate()
+        context = this
         initKoin()
         if (BuildConfig.DEBUG) {
             Timber.plant(Timber.DebugTree())
         }
+        m = BluetoothBeaconIdExchangeManager(this, BeaconIdManager())
+        m.start()
```

Comment:
Moim zdaniem zamiast trzymać tak instancję w klasie Application, lepszym rozwiązaniem będzie przeniesienie jej do Koina. Koin to w uproszczeniu taka wielka mapa, service locator. Można ten obiekt przenieść do modułu koina i tworzyć go w scope `single` co spowoduje, że będzie singletonem. W razie potrzeby w dowolnym miejscu w kodzie można zawołać `KoinContextHandler.get().get< BluetoothBeaconIdExchangeManager >()` który nam zwróci instancję tego obiektu. Byłoby to bardziej spójne z resztą kodu i "czystsze" rozwiązanie :)